xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-macro/src/test/java/org/xwiki/livedata/internal/macro/LiveDataMacroTest.java

```diff
@@ -92,57 +98,153 @@
+        expectedConfig.append("          {'id':'after','name':'after'},".trim());
+        expectedConfig.append("          {'id':'contains','name':'contains'}".trim());
+        expectedConfig.append("        ],".trim());
+        expectedConfig.append("        'dateFormat':'yyyy/MM/dd HH:mm'".trim());
+        expectedConfig.append("      },".trim());
+        expectedConfig.append("      {".trim());
+        expectedConfig.append("        'id':'list',".trim());
+        expectedConfig.append("        'defaultOperator':'contains',".trim());
+        expectedConfig.append("        'operators':[".trim());
+        expectedConfig.append("          {'id':'equals','name':'equals'},".trim());
+        expectedConfig.append("          {'id':'startsWith','name':'startsWith'},".trim());
+        expectedConfig.append("          {'id':'contains','name':'contains'},".trim());
+        expectedConfig.append("          {'id':'empty','name':'empty'}".trim());
+        expectedConfig.append("        ]".trim());
+        expectedConfig.append("      }".trim());
+        expectedConfig.append("    ],".trim());
+        expectedConfig.append("    'defaultFilter':'text',".trim());
+        expectedConfig.append("    'displayers':[".trim());
+        expectedConfig.append("      {'id':'text'},".trim());
+        expectedConfig.append("      {'id':'link'},".trim());
+        expectedConfig.append("      {'id':'html'},".trim());
+        expectedConfig.append("      {'id':'actions'},".trim());
+        expectedConfig.append("      {'id':'boolean'}".trim());
+        expectedConfig.append("    ],".trim());
+        expectedConfig.append("    'defaultDisplayer':'text',".trim());
+        expectedConfig.append("    'pagination':{".trim());
+        expectedConfig.append("      'maxShownPages':10,".trim());
+        expectedConfig.append("      'pageSizes':[15,25,50,100],".trim());
+        expectedConfig.append("      'showEntryRange':true,".trim());
+        expectedConfig.append("      'showNextPrevious':true".trim());
+        expectedConfig.append("    },".trim());
+        expectedConfig.append("    'entryDescriptor':{},".trim());
+        expectedConfig.append("    'actions':[".trim());
+        expectedConfig.append("      {'id':'view','name':'view','icon':{}},".trim());
+        expectedConfig.append("      {'id':'edit','name':'edit','icon':{}},".trim());
+        expectedConfig.append("      {'id':'delete','name':'delete','icon':{'cssClass':'text-danger'}},".trim());
+        expectedConfig.append("      {'id':'copy','name':'copy','icon':{}},".trim());
+        expectedConfig.append("      {'id':'rename','name':'rename','icon':{}},".trim());
+        expectedConfig.append("      {'id':'rights','name':'rights','icon':{}}".trim());
+        expectedConfig.append("    ],".trim());
+        expectedConfig.append("    'selection':{'enabled':false}".trim());
+        expectedConfig.append("  }".trim());
+        expectedConfig.append("}".trim());
```

Comment:
I see a lot of duplication in this JSON between tests. Couldn't you use an actual JSON object (Jackson), initialized with default expected Live Data config before each test, and modified in each test with just the configuration that matters for the test, then serialized as text?

Reply:
I will check